pymongo/asynchronous/srv_resolver.py

```diff
@@ -131,6 +133,11 @@ async def _get_srv_response_and_hosts(
     ) -> tuple[resolver.Answer, list[tuple[str, Any]]]:
         results = await self._resolve_uri(encapsulate_errors)
 
+        if self.__fqdn == results[0].target.to_text():
+            raise ConfigurationError(
+                "Invalid SRV host: return address is identical to SRV hostname"
+            )
```

Comment:
Don't we need to validate this for all results? Can you refactor this to happen in the "Validate hosts" loop below? That way we don't need to call target.to_text() multiple times.

Reply:
Done in f72cf8725f3064fd5293a8920c782044a86e9538, although now I wonder why the test cases passed prior to this change … also did someone miss committing a synchro change?

```
On branch PYTHON-4575
Your branch is up to date with 'origin/PYTHON-4575'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	gridfs/synchronous/__init__.py

```